packages/compose/index.js

```diff
@@ -1,54 +1,54 @@
-var isArray = require('@stamp/is/array');
-var isFunction = require('@stamp/is/function');
-var isObject = require('@stamp/is/object');
-var isStamp = require('@stamp/is/stamp');
-var isComposable = require('@stamp/is/composable');
+'use strict';
 
-var assign = require('@stamp/core/assign');
-var merge = require('@stamp/core/merge');
+const { isArray, isComposable, isFunction, isObject, isStamp } = require('@stamp/is');
+const { assign, merge } = require('@stamp/core');
 
-var slice = Array.prototype.slice;
+const { defineProperties } = Object;
+const { get, set, setPrototypeOf } = Reflect;
 
 /**
  * Creates new factory instance.
  * @returns {Function} The new factory function.
  */
-function createFactory() {
-  return function Stamp(options) {
-    var descriptor = Stamp.compose || {};
+const createFactory = () => {
+  return function Stamp(options = {}, ...args) {
+    const descriptor = Stamp.compose || {};
     // Next line was optimized for most JS VMs. Please, be careful here!
-    var obj = {__proto__: descriptor.methods}; // jshint ignore:line
+    // let obj = { __proto__: descriptor.methods };
+    let obj = {};
+    if (descriptor.methods) setPrototypeOf(obj, descriptor.methods);
 
     merge(obj, descriptor.deepProperties);
     assign(obj, descriptor.properties);
-    Object.defineProperties(obj, descriptor.propertyDescriptors || {});
-
-    if (!descriptor.initializers || descriptor.initializers.length === 0) return obj;
-
-    if (options === undefined) options = {};
-    var inits = descriptor.initializers;
-    var length = inits.length;
-    for (var i = 0; i < length; i += 1) {
-      var initializer = inits[i];
-      if (isFunction(initializer)) {
-        var returnedValue = initializer.call(obj, options,
-          {instance: obj, stamp: Stamp, args: slice.apply(arguments)});
-        obj = returnedValue === undefined ? obj : returnedValue;
-      }
+    defineProperties(obj, descriptor.propertyDescriptors || {});
+
+    // if (!descriptor.initializers || descriptor.initializers.length === 0) return obj;
+    if (descriptor.initializers && descriptor.initializers.length > 0) {
+      const inits = descriptor.initializers;
+      inits.forEach((initializer) => {
```

Comment:
Swithing to a `for (... of ...)` statement

Reply:
Yeah. AFAIK for-of is as performance as for in node 12. Not sure about node 10. And it's time to deprecate node 8 since it's EOL is 1-st of Jan 2020.

Reply:
When dropping NodeJS 8 in January, we'll set target to `es2018` (in either .eslintrc.json or typescript tsconfig.json) though there is little to benefit from it:

- Object methods `entries`, `getOwnPropertyDescriptors` and `values`
- Object.prototype getter/setter methods
- object rest/spread syntax

I have not yet fully investigate the `Proxy` object but it could find its use in a few places like `@stamp/collision` and `@stamp/privatize`